src/wasm/wasm-emscripten.cpp

```diff
@@ -697,10 +776,28 @@ void AsmConstWalker::queueImport(Name importName, std::string baseSig) {
   queuedImports.push_back(std::unique_ptr<Function>(import));
 }
 
+Literal AsmConstWalker::tableIndexForName(Name name) {
+  if (tableIndices.count(name)) {
+    return tableIndices[name];
+  }
+  assert(wasm.table.segments.size() == 1);
```

Comment:
Do we need to assert this here? I don't see how we're relying on wasm.table.segments in this function at all